hydra-node/src/Hydra/API/ServerOutput.hs

```diff
@@ -161,15 +163,46 @@ prepareServerOutput OutputCBOR response =
         Fanout -> encodedResponse
     TxValid{Hydra.API.ServerOutput.transaction = tx} -> replacedResponse tx
     TxInvalid{Hydra.API.ServerOutput.transaction = tx} -> replacedResponse tx
-    SnapshotConfirmed{} -> encodedResponse
+    SnapshotConfirmed{Hydra.API.ServerOutput.snapshot} ->
+      replacedSnapshotTxResponse (confirmed snapshot)
     GetUTxOResponse{} -> encodedResponse
     InvalidInput{} -> encodedResponse
     Greetings{} -> encodedResponse
-    PostTxOnChainFailed{} -> encodedResponse
+    PostTxOnChainFailed{postChainTx} ->
+      case postChainTx of
+        CloseTx{confirmedSnapshot} ->
+          case confirmedSnapshot of
+            InitialSnapshot{} -> encodedResponse
+            ConfirmedSnapshot{Hydra.Snapshot.snapshot} ->
+              replacedSnapshotTxInPostTxOnChainFailedResponse (confirmed snapshot)
+        ContestTx{confirmedSnapshot} ->
+          case confirmedSnapshot of
+            InitialSnapshot{} -> encodedResponse
+            ConfirmedSnapshot{Hydra.Snapshot.snapshot} ->
+              replacedSnapshotTxInPostTxOnChainFailedResponse (confirmed snapshot)
+        _other -> encodedResponse
     RolledBack -> encodedResponse
```

Comment:
As the `key` traversals are optional.. can we just apply them one after another?

i.e. we take the `encode response` and just try to do all the lens updates in sequence. They would only apply if the object has indeed the given shape (e.g. keys `postChainTx.confirmedSnapshot.snapshot.confirmedTransactions` exists)

Reply:
Ah.. the explicit pattern match is what we want to have some sanity instead. Nevermind this suggestion then.